external-import/recorded-future/src/rflib/rf_alerts.py

```diff
@@ -1,389 +1,441 @@
-                        image_presence, image_path, image_name = (
-                            self.api_recorded_future.get_image_and_save_temp_file(
-                                entity["name"]
-                            )
+                    if image_presence:
+                        image_list.append(
+                            {
+                                "name": image_name,
+                                "data": base64.b64encode(image_data),
+                                "mime_type": "image/png",
+                                "objectMarking": self.tlp,
+                            }
                         )
-                        if image_presence:
-                            stix_images.append(
-                                ImageToShowInNote(
-                                    hit["id"],
-                                    image_path,
-                                    image_name,
-                                )
-                            )
-                    entities_string += str(
-                        "entity "
-                        + str(entity_count)
-                        + " - "
-                        + entity["type"]
-                        + " : "
-                        + entity["name"]
-                        + "  \n  "
+                elif entity["type"] == "URL":
+                    stix_url = stix2.URL(
+                        value=entity["name"], object_marking_refs=self.tlp
+                    )
+                    stix_relationship = stix2.Relationship(
+                        relationship_type="related-to",
+                        source_ref=stix_incident.id,
+                        target_ref=stix_url.id,
+                        created_by_ref=self.author,
+                    )
+                    bundle_objects.append(stix_url)
+                    bundle_objects.append(stix_relationship)
+                    value = search(
+                        "(https:\/\/t.me\/.+?(?=\/))|(https:\/\/twitter.com\/.+?(?=\/))",
+                        str(entity["name"]),
+                    )
+                    if value is None:
+                        root_domain = self.get_root_domain(entity["name"])
+                        stix_external_ref = stix2.ExternalReference(
+                            source_name="urlscan.io",
+                            url="https://urlscan.io/domain/" + str(root_domain),
+                        )
+                        stix_external_refs.append(stix_external_ref)
+                        bundle_objects[0] = bundle_objects[0].new_version(
+                            external_references=stix_external_refs
+                        )
+                elif entity["type"] == "IpAddress":
+                    stix_ipv4address = stix2.IPv4Address(
+                        value=entity["name"],
+                        object_marking_refs=self.tlp,
+                        custom_properties={
```

Comment:
replace with: 
```
custom_properties={
    "created_by_ref": self.author["id"]
 }
```
The bug observed on my side lies in the fact that we are not passing an identifier but a dict object. 
This must be added to all observables created. This extension is missing in some observables (e.g. URL creation).